src/utils/utils.ts

```diff
@@ -14,6 +14,20 @@ export const filterFutureAnnouncements = (
   return announcements.filter((announcement) => announcement.startDate > date);
 };
 
+/**
+ * Filters out announcements whose endDate is in the past.
+ *
+ * @param announcements - An array of Announcement objects.
+ * @param date - Optional. The date to compare against each announcement's startDate. If not provided, the current date and time when the function is called will be used.
+ * @returns An array of Announcement objects where each endDate is greater than the current date.
+ */
+export const filterOutPastAnnouncements = (
```

Comment:
I would rename this to `filterActiveAnnouncements`. Then in the docstrings, mention that active announcements are announcements that are either current or upcoming.